File: 5-Architecture design patterns/MVC/Day5_Threads/app/src/main/java/com/example/day5_threads/model/DB/ProductDao.java
package com.example.day5_threads.model.DB;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.day5_threads.model.Pojo.Product;

import java.util.List;

@Dao
public interface ProductDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    public void insertItem(Product product);

    @Delete
    public void deleteItem(Product product);

    @Query("Select * from Product")
    public LiveData<List<Product>> getAllItems();
}

File: 4-Android with Kotlin/Day5/app/src/main/java/com/example/day1/model/local/ProductDao.kt
package com.example.day1.model.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.day1.model.remote.Product

@Dao
interface ProductDao {
    @Query("SELECT * FROM Products_Table")
    suspend fun getAllProducts():List<Product>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insertAllProducts(list :List<Product> )
}